### ISSUE TITLE
Multi level subform

### ISSUE THREAD

**Issue by webchun** [joomla/joomla-cms#11551]
#### Steps to reproduce the issue

![screen shot 2016-08-10 at 19 15 05](https://issues.joomla.org/uploads/1/52c2e59857b9a7db8bfb81f325b03200.gif)
- I'm using subform for my module parameters
- I have this code on my xml : 

```
<field name="items" type="subform" formsource="modules/mod_modulex/fields/item.xml" min="1" max="4" multiple="true" layout="joomla.form.field.subform.repeatable-table" groupByFieldset="true" label="Subform Field" description="Subform Field Description" />
```
- This is the content of the item.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<form>
<fieldset name="field_items" label="Fields">
        <field
            name="field_label"
            type="text"
            label="title"
            size="40" />

        <field name="subchild_items" type="subform" formsource="modules/mod_modulex/fields/subchild_items.xml" max="20" multiple="true" layout="joomla.form.field.subform.repeatable-table" groupByFieldset="true" label="value" description="Checkbox Items" showon="field_type:4"/>
</fieldset>
</form>
```

and this is the content of subchild_items.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<form>
<fieldset name="field_items2" label="Fields">
    <field
        name="subchild_value"
        type="text"
        default="0"
        label="Value"
    />
</fieldset>
</form>
```
#### Expected result

The subform works properly
#### Actual result

It works partially. The subchild subform is displayed, but it's breaking the next fieldset, and every time I added the new item, the subchild subform's field is displayed twice 
#### System information (as much as possible)

Joomla 3.6.2

**Comment by Fedik** [comment]
Multi level subform in `multiple=true` mode does not supported (due some limitation).
However in single (`multiple=false`) mode should work, in theory, but I not tried 😄

**Comment by webchun** [comment]
@Fedik Ah ok ;)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/joomla-cms/11551</a>.</sub>

**Comment by gaelicwinter** [comment]
@Fedik, any idea when the multilevel subform code might be available?  I'm stuck with the need, and while I probably don't have as much experience with jquery as you, I am happy to help in any way I can.  It will probably take me 2-3 weeks to come up with my own solution to use in my own joomla site, and I'd rather devote that to the core project if possible.  I'm good with joomla, php, css, etc, and ok with js and such.

Happy to contribute if I can.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/joomla-cms/11551</a>.</sub>

**Comment by Fedik** [comment]
Some months ago I tried to fix it for my own project, but did not got any success with multilevel subform.
There need some more coding and changes in the script and in the layout.
I am not sure that it can be fixed soon.

**Comment by olejenya** [comment]
showon = "test: 1" This is just the minimum that can be used in multiple = "true", that everything was beautiful.
It would be nice to make this type of field that would work all. The modules have become much brighter and more convenient to the end user ispolbzovanii Joomla!
-------------------------
showon="test:1" Это как раз тот минимум , который можно было использовать в multiple="true" , что бы все было красиво.
Было бы неплохо сделать этот тип поля , что бы все работало. Модули стали бы намного ярче и удобней в исполбзовании для конечного пользователя джумлы<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/joomla-cms/11551</a>.</sub>

**Comment by hennysmafter** [comment]
@Fedik Is there a indication when this is going to be fixed/implemented. Because as of now we can't include any subform from other extensions that have their own subforms. 

multiple=true / multiple=false both give the same results in both cases it does not save any values. See this ticket for a bit more info:https://github.com/joomla/joomla-cms/issues/12838

**Comment by Fedik** [comment]
> Is there a indication when this is going to be fixed/implemented

no idea, this feature even not planned before.
some day, maybe 😄

**Comment by hennysmafter** [comment]
@Fedik So what now how to get this feature on the Agenda?

**Comment by Fedik** [comment]
the topic already opened, need to wait for someone who know how and a have time to fix it

**Comment by brianteeman** [comment]
Please remember that everyone who contributes to Joomla is a volunteer

**Comment by hennysmafter** [comment]
@Fedik Ah oke. As long as it is mentioned it can't be forgotten. As I think it would be nice to have.
@brianteeman I know that and because of them we get awesome features!

**Comment by franz-wohlkoenig** [comment]
@webchun are com_fields a Solution?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/tracker/joomla-cms/11551</a>.</sub>

**Comment by Fedik** [comment]
@franz-wohlkoenig nope

**Comment by franz-wohlkoenig** [comment]
thanks @Fedik, set Status on "Needs Review".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/tracker/joomla-cms/11551</a>.</sub>

**Comment by laoneo** [comment]
This issue can be closed as we have a pr for it #17552.

**Comment by ghost** [comment]
closed as having Pull Request #17552

Thanks for Hint, @laoneo<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11551">issues.joomla.org/tracker/joomla-cms/11551</a>.</sub>
